eth/utils/bitfield.py

```diff
@@ -0,0 +1,41 @@
+from typing import (
+    List,
+)
+
+
+def has_voted(bitfield: bytes, index: int) -> bool:
+    return bool(bitfield[index // 8] & (128 >> (index % 8)))
+
+
+def set_voted(bitfield: bytes, index: int) -> bytes:
+    byte_index = index // 8
+    bit_index = index % 8
+    new_byte_value = bitfield[byte_index] | (128 >> bit_index)
+    return bitfield[:byte_index] + bytes([new_byte_value]) + bitfield[byte_index + 1:]
+
+
+def get_bitfield_length(bit_count: int) -> int:
+    """Return the length of the bitfield for a given number of attesters in bytes."""
+    return (bit_count + 7) // 8
+
+
+def get_empty_bitfield(bit_count: int) -> bytes:
+    return b"\x00" * get_bitfield_length(bit_count)
+
+
+def get_vote_count(bitfield: bytes) -> int:
+    votes = 0
+    for index in range(len(bitfield) * 8):
+        if has_voted(bitfield, index):
+            votes += 1
+    return votes
+
+
+def or_bitfields(bitfields: List[bytes]) -> bytes:
+    new = b''
```

Comment:
A non-mutative approach which depending on desired behavior will require some pre-validation is as follows.  Some behavior which needs to be specified are things like what happens when the various bitfields have different lengths...

Also, the following code will need to be special cased for situations where there are `0` or `1` bitfields.

The following code does the following algorithm.

1. group all of the bytes from each bitfield in a sequence of sequences where the sequence at index `0` is all of the bytes at index `0` from each of the bitfields.
2. do a map reduce using the binary *or* operator on each of the sequences.
3. join the resulting bytes back into a single bitfield

```python
from cytoolz.curried import reduce

byte_slices = zip(*bitfields)
return bytes(map(reduce(operator.or_), byte_slices))
```

Above is untested psuedo code but it should convey the idea.